### PR TITLE
Fixed modsep operator for various rustlang editions

### PR DESCRIPTION
**Pull request by crw5996**
This PR fixes issue #2927

Everything seems pretty straightforward, although I am curious if there is a better way to handle the edition comparison?

### CODE REVIEW

File: src/imports.rs [hunk 1]
```diff
@@ -154,7 +155,13 @@ impl UseSegment {
...
             UseSegment::Super(None)
         } else {
-            UseSegment::Ident((*name).to_owned(), None)
+            if modsep {
+                let mut mod_sep = "::".to_owned();
+                mod_sep.push_str(&(*name).to_owned());
+                UseSegment::Ident(mod_sep, None)
+            } else {
+                UseSegment::Ident((*name).to_owned(), None)
+            }
```
Comment (topecongiro) [inline src/imports.rs]: I prefer to simplify this like the following:

```rust
let mod_sep = if modsep { "::" } else { "" };
UseSegment::Idnet(format!("{}{}", mod_sep, name), None);
```

File: src/imports.rs [hunk 2]
```diff
@@ -315,8 +322,16 @@ impl UseTree {
...
             attrs,
         };
+
+        let mut leading_modsep = false;
+
+        if context.config.edition() == Edition::Edition2018 {
+            if a.prefix.to_string().len() > 2 && &a.prefix.to_string()[0..2] == "::" {
+                leading_modsep = true;
+            }
+        }
```
Comment (topecongiro) [inline src/imports.rs]: I prefer not to use a `mut` variable here. I think that we can simplify this by using `&&` operator instead.

File: src/imports.rs [hunk 3]
```diff
@@ -315,8 +322,16 @@ impl UseTree {
             visibility,
             attrs,
         };
+
+        let mut leading_modsep = false;
+
+        if context.config.edition() == Edition::Edition2018 {
+            if a.prefix.to_string().len() > 2 && &a.prefix.to_string()[0..2] == "::" {
```
Comment (topecongiro) [inline src/imports.rs]: I think that we can use `a.prefix.starts_with("..")` instead of `&a.prefix.to_string()[0..2] == "::"`.

**Comment by topecongiro** [comment]
Sorry, you need to rebase.

**Comment by crw5996** [comment]
Thank you for the comments! Let me know if I need to do anything else

File: src/imports.rs [hunk 4]
```diff
@@ -313,8 +317,18 @@ impl UseTree {
             visibility,
             attrs,
         };
+
+        let leading_modsep = if context.config.edition() == Edition::Edition2018 {
```
Comment (topecongiro) [inline src/imports.rs]: Would you please use `&&` here as well?

```rust
let leading_modsep = context.config.edition() == Edition::Edition2018
    && a.prefix.to_string().len() > 2
    && a.prefix.to_string().starts_with("::");

**Comment by topecongiro** [comment]
@crw5996 Thanks!

**Pull request merged**